frontend/test/validate-e2e-test-files.js

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env node
+const { execSync } = require("child_process");
+
+const path = require("path");
+const glob = require("glob");
+const chalk = require("chalk");
+
+const e2eFileExtension = ".cy.spec.js";
+const e2eHome = "frontend/test/metabase/scenarios/";
+
+init();
+
+function validateStagedFiles() {
+  const stagedFiles = execSync("git diff HEAD --name-only --diff-filter=d", {
+    encoding: "utf8",
+  });
+  const relevantStagedFiles = stagedFiles.includes(e2eHome);
+  if (!relevantStagedFiles) {
+    return;
+  } else {
+    const invalidFileNames = stagedFiles
+      .split("\n")
+      .filter(fullPath => {
+        const dirName = path.dirname(fullPath);
+        const excludedPaths =
+          dirName.endsWith("/helpers") || dirName.endsWith("/shared");
+        return dirName.startsWith(e2eHome) && !excludedPaths;
+      })
```

Comment:
do you think abstract this function into its own with a name would make this easier to read? e.g.
```suggestion
      .filter(isE2ETestFile)
```